### PR TITLE
Fixed user renamed

### LINKED ISSUE #2337: Change of name is not working properly
We have a possibility to change the name, but currently it works incorrectly.
We are simply redirected to a default view without any authorization due to an incorrect updating of UserInfo

### PR DESCRIPTION
**Pull request by Cheshiriks**
Fixed user renamed

Closes #2337

### CODE REVIEW

**Review by sanyavertolet — APPROVED** [review]

File: authentication-service/src/main/kotlin/com/saveourtool/save/authservice/security/ConvertingAuthenticationManager.kt [hunk 1]
```diff
@@ -33,7 +33,8 @@ class ConvertingAuthenticationManager(
      */
     override fun authenticate(authentication: Authentication): Mono<Authentication> = if (authentication is UsernamePasswordAuthenticationToken) {
         val (name, identitySource) = authentication.extractUserNameAndIdentitySource()
-        authenticationUserDetailsService.findByUsername(name)
+        val nameAndSource = "$name@SAVE@$identitySource"
```
Comment (kgevorkyan) [inline authentication-service/src/main/kotlin/com/saveourtool/save/authservice/security/ConvertingAuthenticationManager.kt:36]: separator should be extracted into single variable into common part

File: authentication-service/src/main/kotlin/com/saveourtool/save/authservice/repository/AuthenticationUserRepository.kt [hunk 2]
```diff
@@ -13,17 +13,22 @@ class AuthenticationUserRepository(
     private val namedParameterJdbcTemplate: NamedParameterJdbcTemplate,
 ) {
     /**
-     * @param name
+     * @param userNameAndSource
      * @return user or null if no results have been found
      */
-    fun findByName(name: String): User? {
+    fun findByName(userNameAndSource: String): User? {
```
Comment (nulls) [inline authentication-service/src/main/kotlin/com/saveourtool/save/authservice/repository/AuthenticationUserRepository.kt:19]: suggestion to rename the method or pass two arguments here

**Review by akuleshov7 — APPROVED** [review]

File: authentication-service/src/main/kotlin/com/saveourtool/save/authservice/repository/AuthenticationUserRepository.kt (hunk 2 above)
Comment (nulls) [reply authentication-service/src/main/kotlin/com/saveourtool/save/authservice/repository/AuthenticationUserRepository.kt:19]: `com.saveourtool.save.authservice.service.AuthenticationUserDetailsService` is from Spring. 
`AuthenticationUserRepository.kt` is our repository class

File: authentication-service/src/main/kotlin/com/saveourtool/save/authservice/repository/AuthenticationUserRepository.kt [hunk 3]
```diff
@@ -13,17 +13,22 @@ class AuthenticationUserRepository(
...
         ).singleOrNull()
-            .orNotFound {
-                "There is no user with name $name"
-            }
+            ?: namedParameterJdbcTemplate.queryForList(
+                "SELECT * FROM save_cloud.user WHERE id = (select user_id from save_cloud.original_login where name = :name AND source = :source)",
+                mapOf("name" to name)
+            ).singleOrNull()
+                .orNotFound {
+                    "There is no user with name $name"
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply authentication-service/src/main/kotlin/com/saveourtool/save/authservice/repository/AuthenticationUserRepository.kt]: Fixed

File: authentication-service/src/main/kotlin/com/saveourtool/save/authservice/security/ConvertingAuthenticationManager.kt (hunk 1 above)
Comment (Cheshiriks) [reply authentication-service/src/main/kotlin/com/saveourtool/save/authservice/security/ConvertingAuthenticationManager.kt]: Fixed

File: authentication-service/src/main/kotlin/com/saveourtool/save/authservice/repository/AuthenticationUserRepository.kt [hunk 4]
```diff
@@ -13,17 +13,22 @@ class AuthenticationUserRepository(
     private val namedParameterJdbcTemplate: NamedParameterJdbcTemplate,
 ) {
     /**
-     * @param name
+     * @param userNameAndSource
      * @return user or null if no results have been found
      */
-    fun findByName(name: String): User? {
+    fun findByName(userNameAndSource: String): User? {
+        val (name, source) = userNameAndSource.split("@SAVE@")
```
[the remark this answers is not in the document]
Comment (Cheshiriks) [reply authentication-service/src/main/kotlin/com/saveourtool/save/authservice/repository/AuthenticationUserRepository.kt]: Fixed

File: authentication-service/src/main/kotlin/com/saveourtool/save/authservice/repository/AuthenticationUserRepository.kt (hunk 2 above)
Comment (Cheshiriks) [reply authentication-service/src/main/kotlin/com/saveourtool/save/authservice/repository/AuthenticationUserRepository.kt]: Changed it

File: authentication-service/src/main/kotlin/com/saveourtool/save/authservice/security/ConvertingAuthenticationManager.kt [hunk 5]
```diff
@@ -62,4 +63,8 @@ class ConvertingAuthenticationManager(
                     identitySource = identitySource,
                 )
             }
+
+    companion object Factory {
+        const val AUTH_SEPARATOR = "@SAVE@"
```
Comment (kgevorkyan) [inline authentication-service/src/main/kotlin/com/saveourtool/save/authservice/security/ConvertingAuthenticationManager.kt:68]: I think all constants should be in our Constants.kt

**Review by kgevorkyan — APPROVED** [review]

File: authentication-service/src/main/kotlin/com/saveourtool/save/authservice/security/ConvertingAuthenticationManager.kt (hunk 5 above)
Comment (Cheshiriks) [reply authentication-service/src/main/kotlin/com/saveourtool/save/authservice/security/ConvertingAuthenticationManager.kt]: Fixed

**Pull request merged**